tests/foreman/api/test_template.py

```diff
@@ -112,6 +113,43 @@ def test_negative_create_with_invalid_name(self):
                     entities.ConfigTemplate(name=name).create()
 
     @tier1
+    def test_positive_create_with_template_kind(self):
+        """Create a provisioning template providing the template_kind.
+
+        @id: d7309be8-b5c9-4f77-8c4e-e9f2b8982076
+
+        @Assert: Provisioning Template is created and contains provided
+        template kind.
+        """
+        template_kind = entities.TemplateKind().search(
+            query={'search': 'name="PXELinux"'})[0]
+        template = entities.ProvisioningTemplate(
+            snippet=False, template_kind=template_kind,
+        ).create()
+        self.assertEqual(template.template_kind.id, template_kind.id)
+
+    @tier1
+    def test_negative_create_with_template_kind_name(self):
+        """Create a provisioning template providing non-existing
+        template_kind name.
+
+        @id: e6de9ceb-fe4b-43ce-b7e3-5453ca4bd164
+
+        @Assert: 404 error and expected message is returned
+
+        @BZ: 1379006
+        """
+        template = entities.ProvisioningTemplate(snippet=False)
+        template.template_kind_name = gen_string('alpha')
```

Comment:
So what happens in case of correct template_kind_name (but no id specified)? Can't see this case covered here. If it doesn't work with correct name passed - then basically this test makes no sense as functionality only works with id (which i guess would be a bug which needs to be resolved).